web/console/src/app/minting/service.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2022 Creditor Corp. Group.
+// See LICENSE for copying information.
+
+import MetaMaskOnboarding from '@metamask/onboarding';
+
+import { metamaskNotifications } from '@/app/internal/notifications';
+import { ServicePlugin } from '@/app/plugins/service';
+
+import CasperTransactionService from '@/app/casper';
```

Comment:
service in service?
if you can have multiple wallets
perhaps you need to write an interface for Wallet
and implement it separately for different wallets
then write a shared service
and call it in view?
Example here: https://github.com/BoostyLabs/golden-gate-connector-casper/pull/43

Reply:
Because there are a lot of rows of code in the related to minting components, so i decided to separate all the logic with minting in service

Reply:
Check plz logic here https://github.com/BoostyLabs/golden-gate-web-app/pull/98
We have:
1) common Wallet interface 
2) metamaskWAllet, casperWalett, etc 
3) Common WalletService 

You have: 
A lot of services (CasperTransactionService, ServicePlugin, minting service)
I'm not sure if you can do this (slightly different logic)
but worth a try

Reply:
I am not into this repo